contracts/Exchange.sol

```diff
@@ -678,9 +680,9 @@ contract Exchange is SafeMath, Destructable {
     /// @return Allowance of token given to TokenTransferProxy by owner.
     function getAllowance(address token, address owner)
         internal
-        view  // The called token contract may attempt to change state, but will not be able to due to an added gas limit.
         returns (uint)
     {
-        return Token(token).allowance.gas(EXTERNAL_QUERY_GAS_LIMIT)(owner, this); // Limit gas to prevent reentrancy
+        // The called token contract may attempt to change state, but will not be able to due to an added gas limit.
```

Comment:
This comment should stay outside of the function, as it is documentation for the function caller and not a comment how the function works. 
Try to include it in the docs block maybe?